sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -0,0 +1,627 @@
+package com.mapbox.maps
+
+import android.os.Handler
+import android.os.Looper
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import com.mapbox.bindgen.Value
+import com.mapbox.common.*
+import com.mapbox.geojson.Point
+import com.mapbox.maps.extension.observable.getResourceEventData
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.*
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class OfflineTest {
+
+  private val handler = Handler(Looper.getMainLooper())
+  private val offlineManager: OfflineManager by lazy {
+    OfflineManager(
+      MapInitOptions.getDefaultResourceOptions(
+        InstrumentationRegistry.getInstrumentation().targetContext
+      )
+    )
+  }
+  private val cacheManager: CacheManager by lazy {
+    CacheManager(
+      MapInitOptions.getDefaultResourceOptions(
+        InstrumentationRegistry.getInstrumentation().targetContext
+      )
+    )
+  }
+
+  @get:Rule
+  var rule = ActivityScenarioRule(EmptyActivity::class.java)
+
+  @Before
+  fun setUp() {
+    val latch = CountDownLatch(2)
+    handler.post {
+      offlineManager.removeStylePack(STYLE)
+      TileStore.getInstance().removeTileRegion(TILE_REGION_ID)
+      TileStore.getInstance().setOption(TileStoreOptions.DISK_QUOTA, Value(0))
+      cacheManager.apply {
+        // clear cache in a most severe way
+        clearAmbientCache {
+          latch.countDown()
+        }
+        // restore ambient cache size to some default value > 0
+        setMaximumAmbientCacheSize(DEFAULT_AMBIENT_CACHE_SIZE_BYTES) {
+          latch.countDown()
+        }
+      }
+    }
+    if (!latch.await(10, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun loadStylePackNoCancel() {
+    val latch = CountDownLatch(1)
+    var downloadProgress: StylePackLoadProgress? = null
+    var downloadedStyle: StylePack? = null
+    var downloadedStyleError: StylePackError? = null
+    handler.post {
+      offlineManager.loadStylePack(
+        STYLE,
+        StylePackLoadOptions.Builder()
+          .glyphsRasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
```

Comment:
you will need to specify a fontFamily here.

Reply:
can you provide more information what does it mean in this case? where should I specify `fontFamily` here?

Reply:
oh, I am referring to something like https://github.com/mapbox/mapbox-maps-android/blob/5ad0667774d62e72b43cc469ca86b4f30521c7b8/sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt#L51-L54 , but it seems we don't have an option to specify it here. @pozdnyakov do you think we need an option here to specify the text fonts?

Reply:
Nope, font family is not an option for offline download but only for the map (for the local glyph rasterizer)